src/AutoRest.CSharp/Mgmt/Decorator/CodeWriterExtensions.cs

```diff
@@ -41,24 +42,23 @@ public static CodeWriter WriteXmlDocumentationMgmtRequiredParametersException(th
                 var nonEmptyParameters = requiredParameters.Where(p => HasEmptyCheck(p)).ToArray();
                 if (nonEmptyParameters.Any())
                 {
-                    var nullOrEmptyDescription = GetExceptionDescription(nonEmptyParameters, true);
-                    writer.WriteXmlDocumentationException(typeof(ArgumentException), nullOrEmptyDescription);
+                    var nonEmptyDescription = GetExceptionDescription(nonEmptyParameters, "empty");
+                    writer.WriteXmlDocumentationException(typeof(ArgumentException), nonEmptyDescription);
                 }
 
-                var nonNullParameters = requiredParameters.Except(nonEmptyParameters).ToArray();
+                var nonNullParameters = requiredParameters;
                 if (nonNullParameters.Any())
                 {
-                    var nullDescription = GetExceptionDescription(nonNullParameters, false);
+                    var nullDescription = GetExceptionDescription(nonNullParameters, "null");
                     writer.WriteXmlDocumentationException(typeof(ArgumentNullException), nullDescription);
                 }
             }
 
             return writer;
         }
 
-        private static FormattableString GetExceptionDescription(IReadOnlyList<Parameter> parameters, bool canBeEmpy)
+        private static FormattableString GetExceptionDescription(IReadOnlyList<Parameter> parameters, string message)
```

Comment:
`message` is more like `errorDescription`?

Reply:
well, it is not a full message... how about I rename it to just `nullOrEmpty` because it can only be `null` or `empty`